site/app/controllers/MiscController.php

```diff
@@ -24,59 +24,110 @@ public function run() {
-            if ($part == ".." || $part == ".") {
-                throw new \InvalidArgumentException("Cannot have a part of the path just be dots");
+    // function to check that this is a valid access request
+    private function checkValidAccess($is_zip, $dir="",$path="",$user_id="") {
+
+        // if not a zip
+        if (!$is_zip) {
+            // first check that path is valid 
+            foreach (explode(DIRECTORY_SEPARATOR, $path) as $part) {
+                if ($part == ".." || $part == ".") {
+                    throw new \InvalidArgumentException("Cannot have a part of the path just be dots");
+                }
             }
-        }
-        $path = $this->core->getConfig()->getCoursePath();
-        if ($_REQUEST['dir'] === "config_upload") {
-            $check = FileUtils::joinPaths($path, "config_upload");
-            if (!Utils::startsWith($_REQUEST['path'], $check)) {
-                throw new \InvalidArgumentException("Path must start with path to config_upload");
+            $course_path = $this->core->getConfig()->getCoursePath();
+            if ($dir === "config_upload") {
+                $check = FileUtils::joinPaths($course_path, "config_upload");
+                if (!Utils::startsWith($path, $check)) {
+                    throw new \InvalidArgumentException("Path must start with path to config_upload");
+                }
+                if (!file_exists($path)) {
+                    throw new \InvalidArgumentException("File does not exist");
+                }
             }
-            if (!file_exists($_REQUEST['path'])) {
-                throw new \InvalidArgumentException("File does not exist");
+            else if ($dir === "submissions" || $dir === "uploads") {
+                if (!file_exists($path)) {
+                    throw new \InvalidArgumentException("File does not exist");
+                }
             }
-        }
-        else if ($_REQUEST['dir'] === "submissions" || $_REQUEST['dir'] === "uploads") {
-            if (!file_exists($_REQUEST['path'])) {
-                throw new \InvalidArgumentException("File does not exist");
+            else {
+                throw new \InvalidArgumentException("Invalid dir used");
+            }
+            if (!FileUtils::isValidFileName($path)) {
+                throw new \InvalidArgumentException("Not a valid file name");
+            }
+            $folder_names = array();
+            $folder_names[] = "submissions";
+            $folder_names[] = "results";
+            $folder_names[] = "checkout";
+            $folder_names[] = "uploads";
+            $path_anchors = array();
+            $path_anchors[] = FileUtils::joinPaths($course_path, $folder_names[0]);
+            $path_anchors[] = FileUtils::joinPaths($course_path, $folder_names[1]);
+            $path_anchors[] = FileUtils::joinPaths($course_path, $folder_names[2]);
+            $path_anchors[] = FileUtils::joinPaths($course_path, $folder_names[3]);
```

Comment:
In looking at this again, it doesn't seem like there's any point in setting up these arrays as we use them immediately and then never again so just construct them on the fly for the check.
```
$folder_names = array();
foreach (array('submissions', 'results', 'checkout', 'uploads') as $folder) {
    if (Utils::startsWith($path, FileUtils::joinPaths($course_path, $folder))) {
        $access = true;
        break;
    }
}
```